services/admin/src/store.ts

```diff
@@ -223,22 +277,15 @@ export class Store {
     filePath: string;
     originalFilename?: string;
     analyzeOnly?: boolean;
-  }): Promise<
-    Result<
-      {
-        id: Id;
-        wasExistingFile: boolean;
-        newlyAdded: number;
-        alreadyPresent: number;
-      },
-      AddCastVoteRecordError
-    >
-  > {
-    this.client.run('begin transaction');
-    let inTransaction = true;
+  }): Promise<AddCastVoteResult> {
+    const election = this.getElection(electionId)?.electionDefinition.election;
+    if (!election) {
+      return err({ kind: 'InvalidElectionId' });
+    }
+
     const sha256Hash = await sha256File(filePath);
 
-    try {
+    const transactionFn = async (): Promise<AddCastVoteResult> => {
```

Comment:
Use function declaration style?
```suggestion
    async function transactionFn(): Promise<AddCastVoteResult> {
```

Reply:
i assumed that would cause the references to `this` within the function to be overridden, but if we have a strong preference for function dec, i could maybe wrap it in a call and pass in the parent reference?

Reply:
Ah, yes it would. Didn't notice the `this` in there. Our linters prefer function declaration but allow for arrow when `this` is used.